stats/view/view_test.go

```diff
@@ -371,3 +371,31 @@ func containsRow(rows []*Row, r *Row) bool {
 	}
 	return false
 }
+
+func TestRegisterUnregisterParity(t *testing.T) {
```

Comment:
What does this test case test?

Reply:
It tests that invoking register then unregister n number of times results in a full deregistration and never errors like it is currently doing before this PR. It is basically the reproducer for bug #710